docs/Golden-Gate/Bridge-Walkthrough.rst

```diff
@@ -0,0 +1,144 @@
+definition. Here we have to process tokens generated by the target, and
+expose a memory-mapped interface to the bridge driver.
+
+Inspecting the top of the class:
+
+.. literalinclude:: ../../sim/firesim-lib/src/main/scala/bridges/UARTBridge.scala
+    :language: scala
+    :start-after: DOC include start: UART Bridge Header
+    :end-before: DOC include end: UART Bridge Header
+
+
+Most of what follows is responsible for modeling the timing of the UART.
+As a bridge designer, you're free to take as many host-cycles as you need to
+process tokens. In simpler models, like this one, it's often easiest to write
+logic that operates in a single cycle but gate state-updates using a
+"fire" signal that is asserted when the required tokens are available.
+
+Now, we'll skip to the end to see how to add registers to the simulator's memory map
+that can be accessed using MMIO from bridge driver.
+
+.. literalinclude:: ../../sim/firesim-lib/src/main/scala/bridges/UARTBridge.scala
+    :language: scala
+    :start-after: DOC include start: UART Bridge Footer
+    :end-before: DOC include end: UART Bridge Footer
+
+
+UART Bridge Driver
+++++++++++++++++++
+
+To complete our host-side definition, we need to define a CPU-hosted bridge driver.
+Bridge Drivers extend the ``bridge_driver_t`` interface, which declares 5 virtual methods
+a concrete bridge drive must implement. The declaration of the Uart bridge
+driver lives at ``sim/firesim-lib/src/main/cc/bridges/uart.h``. It is inlined
+below:
+
+.. include:: ../../sim/firesim-lib/src/main/cc/bridges/uart.h
+   :code: c++
+
+The bulk of the driver's work is done in its ``tick()`` method. Here, the driver
+polls the BridgeModule and then does some work. Note: the name, ``tick`` is vestigial: one
+invocation of tick() many do work corresponding to an arbitrary number of
+target cycles. It's critical that tick be non-blocking, as waiting for work
+from the BridgeModule may deadlock the simulator.
+
+Adding the BridgeDriver to the Simulator
+++++++++++++++++++++++++++++++++++++++++
+
+With the Bridge Driver implemented, we can register it in main simulator class defined in
+``sim/src/main/cc/firesim/firesim_top.cc``.
+
+.. literalinclude:: ../../sim/src/main/cc/firesim/firesim_top.cc
+    :language: c++
+    :start-after: DOC include start: UART Bridge Driver Registration
+    :end-before: DOC include end: UART Bridge Driver Registration
+
+Build System Modifications
+++++++++++++++++++++++++++
+
+The final consideration in adding your bridge concerns the build system. You
+should be able to host the Scala sources for your bridge with rest of your
+target RTL: SBT will make sure those classes are available on the runtime
+classpath. If you're hosting your bridge driver sources outside of the existing
+directories, you'll need to modify your target-project Makefrag to include them. The default
+Chipyard/Rocket Chip-based one lives here:
+``src/main/makefrag/firesim/Makefrag``
```

Comment:
This is in the firesim docs, so you might want to make the path relative to firesim?